api/turing/service/experiment_service_test.go

```diff
@@ -8,18 +8,141 @@ import (
 	"testing"
 	"time"
 
-	"github.com/gojek/turing/engines/experiment/manager"
-	"github.com/gojek/turing/engines/experiment/manager/mocks"
-	"github.com/gojek/turing/engines/experiment/pkg/request"
 	"github.com/patrickmn/go-cache"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+
+	"github.com/gojek/turing/engines/experiment/manager"
+	"github.com/gojek/turing/engines/experiment/manager/mocks"
+	"github.com/gojek/turing/engines/experiment/pkg/request"
 )
 
 var standardExperimentManagerConfig = manager.Engine{Type: manager.StandardExperimentManagerType}
 var customExperimentManagerConfig = manager.Engine{Type: manager.CustomExperimentManagerType}
 
+func TestIsStandardExperimentManager(t *testing.T) {
+	tests := map[string]struct {
+		engineInfo    manager.Engine
+		engineInfoErr string
+		expected      bool
+	}{
+		"standard": {
+			engineInfo: manager.Engine{
+				Name: "standard-engine-1",
+				Type: manager.StandardExperimentManagerType,
+			},
+			expected: true,
+		},
+		"error": {
+			engineInfo: manager.Engine{
+				Name: "standard-engine-2",
+				Type: manager.StandardExperimentManagerType,
+			},
+			engineInfoErr: "test error",
+		},
+		"custom": {
+			engineInfo: manager.Engine{
+				Name: "custom-engine",
+				Type: manager.CustomExperimentManagerType,
+			},
+		},
+	}
+
+	for name, data := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Set up mock API call
+			expMgr := &mocks.ExperimentManager{}
+			var err error
+			if data.engineInfoErr != "" {
+				err = errors.New(data.engineInfoErr)
+			}
+			expMgr.On("GetEngineInfo").Return(data.engineInfo, err)
+			// Set up experiment service
+			svc := &experimentsService{
+				experimentManagers: map[string]manager.ExperimentManager{
+					data.engineInfo.Name: expMgr,
+				},
+				cache: cache.New(time.Second, time.Second),
+			}
+			// Validate
+			isStdEngine := svc.IsStandardExperimentManager(data.engineInfo.Name)
+			assert.Equal(t, data.expected, isStdEngine)
+		})
+	}
+}
+
+func TestIsClientSelectionEnabled(t *testing.T) {
+	// Set up mock experiment managers
+	tests := map[string]struct {
+		engineInfo    manager.Engine
+		engineInfoErr string
+		expectedErr   string
+		expected      bool
+	}{
+		"standard | no client selection": {
+			engineInfo: manager.Engine{
+				Name: "standard-engine-1",
+				Type: manager.StandardExperimentManagerType,
+				StandardExperimentManagerConfig: &manager.StandardExperimentManagerConfig{
+					ClientSelectionEnabled: false,
```

Comment:
nit: Any value in adding these testcases too?
- `&manager.StandardExperimentManagerConfig{}` for `engineInfo.StandardExperimentManagerConfig.ClientSelectionEnabled` condition
- `StandardExperimentManagerConfig: nil` for `engineInfo.StandardExperimentManagerConfig != nil` condition

Reply:
* `&manager.StandardExperimentManagerConfig{}` is the same as `&manager.StandardExperimentManagerConfig{ClientSelectionEnabled: false}` - so I don't see any use for this.
* `StandardExperimentManagerConfig: nil` - I initially decided not to bother about this because it's only precautionary (`NewExperimentsService` will ensure that a `nil` value is not possible for this field if the type if determined to be standard). But I guess it's good to cover nonetheless - I'll add it in. Thanks for the feedback!